src/core_read.cpp

```diff
@@ -129,18 +129,22 @@ bool DecodeHexTx(CMutableTransaction& tx, const std::string& strHexTx, bool fTry
         }
     }
 
-    CDataStream ssData(txData, SER_NETWORK, PROTOCOL_VERSION);
-    try {
-        ssData >> tx;
-        if (!ssData.empty()) {
+    if (try_witness) {
+        CDataStream ssData(txData, SER_NETWORK, PROTOCOL_VERSION);
+        try {
+            ssData >> tx;
+            if (!ssData.empty()) {
+                return false;
```

Comment:
Even shorter: invert this test and make it return true.

Reply:
Nice :+1: